src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs

```diff
@@ -270,6 +270,15 @@ public static int WriteMetricName(byte[] buffer, int cursor, string metricName,
             return cursor;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int WriteEof(byte[] buffer, int cursor)
+        {
+            cursor = WriteAsciiStringNoEscape(buffer, cursor, "# EOF");
```

Comment:
I don't know enough about prometheus, but I assume this won't cause an older prometheus scrapper to blow up?

Also, is simply adding `# EOF` enough to warrant a change to the content-type to `application/openmetrics-text`?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/a87f8f7373b13dad994e5a9414aed356ec712f67/src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusExporterMiddleware.cs#L76

Reply:
> I don't know enough about prometheus, but I assume this won't cause an older prometheus scrapper to blow up?

I've tested this with older version of scrappers.

![image](https://user-images.githubusercontent.com/17327289/190017257-d6855c87-ff74-4025-af2a-b62d966fb033.png)

> Also, is simply adding `# EOF` enough to warrant a change to the content-type to `application/openmetrics-text`?
> 
> https://github.com/open-telemetry/opentelemetry-dotnet/blob/a87f8f7373b13dad994e5a9414aed356ec712f67/src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusExporterMiddleware.cs#L76

I guess the answer is no, I keep learning new things while reading the OpenMetrics spec, e.g. https://github.com/open-telemetry/opentelemetry-dotnet/pull/3651. I think we need to do a full scan of the spec, maybe it's okay to flip the content-type earlier?

Reply:
> maybe it's okay to flip the content-type earlier?

Probably fine to wait since content-type might be a thing that would affect older scrapers. But may become necessary with exemplar support?